### ISSUE TITLE
sprite-map() magic variables don't support filenames with dashes.

### ISSUE THREAD

**Issue by heygrady** [Compass/compass#898]
This is not the same as #828. When using the magic variables to set options for individual sprites in the map the sprite map variable or the image itself will not work if they contain a dash in the name. This was tested against master.

**Example:** I have a folder named "my-sprites" with a file inside named "foo-bar.png" and I want to capture the map as `$my-sprites`

``` scss
$my-sprites: sprite-map("my-sprites/*.png", $my-sprites-foo-bar-spacing: 10px); // spacing will not have any effect
```

What's happening is that the options are getting all of the dashes replaced with underscores. For instance, "$my-sprites-foo-bar-spacing" becomes "my_sprites_foo_bar_spacing" when it is used.

https://github.com/chriseppstein/compass/blob/stable/lib/compass/sass_extensions/sprites/image.rb#L57

Where the image options are being parsed, the option will never be found because it is searching for the wrong values.

``` rb
def get_var_file(var)
  puts "#{base.name}_#{name}_#{var}" #--> my-sprites_foo-bar_spacing
  # my_sprites_foo_bar_spacing will never be found
  options.get_var "#{base.name}_#{name}_#{var}"
end
```

The problem is that `base.name` and `name` still contain dashes while in the passed options the dashes have already been converted to underscores.

**Comment by scottdavis** [comment]
would you mind creating a failing test and ill get it fixed?

**Comment by chriseppstein** [comment]
In Sass, all variables are normalized and stored internally with dashes converted to underscores. So the `get_var` function [normalizes the access](https://github.com/chriseppstein/compass/blob/stable/lib/compass/sass_extensions/functions/sprites.rb#L9).

Can you either send a patch with tests, or provide a reproducible set of steps to reproduce the issue.

**Comment by heygrady** [comment]
Maybe this can be closed. I was really confused trying to guess how the new variable prefixing works with the undocumented changes to sprites since version 0.12.

I was able to track it down that the variable it comes up is not corrected with underscores, like I showed above. This is still the case. For instance, an image named "foo-bar.png" in the folder named "some-sprites" will cause the `get_var_file` function to look for a variable named `some-sprites_foo-bar_spacing` when the sprite is generated.

However, it appears that it's still finding the all underscore key that is set on the options array. I was getting confused what the prefix should actually be. But now it's clear to me that the prefix is the folder itself. At the time I was able to get it to work by renaming my files to use an underscore. I can't seem to reproduce now. I must've been testing too many variables at once.

Also, if the folder is nested a few folders deep, only the last folder is used for the prefix. For instance, sprites in the folder, "several/folders/deep/" will need variables prefixed only with "deep". For instance, `$deep-foo-bar-spacing`.

I still don't know why there was a BC breaking prefix added to the code but not documented anywhere except in a few bug reports. I don't really know what advantage prefixing `base.name` to the variable solves except to confuse users and break older projects.
